Add script for pushing deb and rpm packages to artifactory

Felixoid opened the pull request:
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Detailed description / Documentation draft:
The script downloads artifacts for a given build and pushes them to the artifactory. Most probably, the workflow will be changed here too.

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The script for uploading packages to the artifactory is added.

Relates to #31644

> By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

Felixoid committed:
Add script for pushing deb and rpm packages to artifactory

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=31748) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=31748) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=31748) it.</sub>

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+    env = os.getenv(name, default)
+    if env is not None:
+        return env
+    raise KeyError("necessary {} environment is not set".format(name))
```
```suggestion
    raise KeyError(f"Necessary {name} environment is not set")
```

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+        self.packages = Packages(version)
+
+    @staticmethod
+    def get_with_retries(url) -> requests.Response:
```
BTW, I have this function https://github.com/ClickHouse/ClickHouse/blob/27ff9d1fc71beb94332e431c5ed9e9123e31d73d/tests/ci/build_download_helper.py#L28-L64, maybe it can be useful here.

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+
+    def deploy_deb(self, packages: Packages):
+        for package in packages.deb:
+            self.deb(package).deploy_deb(
```
Nice API :)

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+        args.build_name,
+        args.release.version,
+    )
+    artClient = Artifactory(args.artifactory_url, args.release.type)
```
```suggestion
    art_client = Artifactory(args.artifactory_url, args.release.type)
```

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+        # "33333/" or "0/"
+        "{pr}/"
+        # "2bef313f75e4cacc6ea2ef2133e8849ecf0385ec/"
+        "{commit}/"
```
I hope we will fix it soon.

Felixoid replied:
What exactly? Do we want to have another bucket template for build's artifacts?

alesapin replied:
Just get rid of `clang-13_relwithdebuginfo_memory_bundled_unsplitted_notidy_without_coverage_deb`

alesapin commented on utils/release/push_to_artifactory:
```diff
@@ -0,0 +1,263 @@
+        elif JFROG_TOKEN:
+            path = ArtifactorySaaSPath(url, token=JFROG_TOKEN)
+        else:
+            raise KeyError("neither JFROG_API_KEY nor JFROG_TOKEN env are defined")
```
```suggestion
            raise KeyError("Neither JFROG_API_KEY nor JFROG_TOKEN env are defined")
```

alesapin reviewed (approved):
LGTM

Felixoid committed:
Apply suggestions from code review

Co-authored-by: alesapin <alesapin@clickhouse.com>

Felixoid committed:
Finish the script

- add logging to rpm and deb deployment
- fix art_client name
- use dowload_build_with_progress function
- move the script to tests/ci

Felixoid commented:
It's tested and works. I merge it and create another PR with CI workflow
